src/ResourceManager/Websites/Commands.Websites/Cmdlets/DeploymentSlots/SwapAzureWebAppSlot.cs

```diff
@@ -0,0 +1,111 @@
+﻿
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+
+using Microsoft.Azure.Commands.WebApps.Utilities;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.WebApps.Cmdlets.DeploymentSlots
+{
+    /// <summary>
+    /// this commandlet will let you swap two web app slots using ARM APIs
+    /// </summary>
+    [Cmdlet("Swap", "AzureRmWebAppSlot", SupportsShouldProcess = true)]
+    public class SwapAzureWebAppSlot : WebAppBaseCmdlet
+    {
+        [Parameter(Position = 0, Mandatory = true, HelpMessage = "Name of the source slot.", ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string SourceSlotName { get; set; }
+
+        [Parameter(Position = 1, Mandatory = false, HelpMessage = "Name of the destination slot.")]
+        [ValidateNotNullOrEmpty]
+        public string DestinationSlotName { get; set; }
+
+        [Parameter(Position = 2, Mandatory = false, HelpMessage = "Swap with preview action to use", ValueFromPipelineByPropertyName = true)]
```

Comment:
Yes, it doesn't make sense to accept this property from the pipeline as our cmdlets are designed to accept web app object from the pipeline instead. I have removed this attribute.